### PR TITLE
Fixing spotbugs errors in all files related 3.6.3 dev branch

### PR DESCRIPTION
**Pull request by rahulrane50**
mvn spotbugs has reported few bugs in these files. Most of the bugs have been fixed in this PR and few are suppressed.

### CODE REVIEW

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java [hunk 1]
```diff
@@ -267,6 +273,11 @@ public String getZnodeGroupAclSuperUserId() {
     return znodeGroupAclSuperUserId;
   }
 
+  /**
+   * crossDomainAccessDomains are loaded lazily and hence spotbugs DC_DOUBLECHECK warning is supressed.
+   * @return
+   */
+  @SuppressFBWarnings("DC_DOUBLECHECK")
```
Comment (desaikomal) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java:280]: any particular reason, you are suppressing vs/ fixing?

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java [hunk 2]
```diff
@@ -148,9 +149,12 @@ public boolean isValid(String id) {
   }
 
   /**
-   * Initialize a new ClientUriDomainMappingHelper instance if it hasn't been instantiated for the ACL provider.
+   * Initialize a new ClientUriDomainMappingHelper instance lazily if it hasn't been instantiated for the ACL provider.
+   * Since it's lazy initialization suppressing spotbugs DC_DOUBLECHECK warning.
```
Comment (desaikomal) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java:153]: is this true for all lazy initializations?

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java [hunk 3]
```diff
@@ -122,7 +122,10 @@ public BackupFile(File backupFile, boolean isTemporaryFile, ZxidRange zxidRange)
      */
     public void cleanup() {
       if (isTemporary && exists()) {
-        file.delete();
+        boolean ret = file.delete();
```
Comment (desaikomal) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/BackupManager.java:125]: just a nit: if (!file.delete())  can directly be used.

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorage.java [hunk 4]
```diff
@@ -166,11 +168,15 @@ public void copyToBackupStorage(File srcFile, File destName) throws IOException
...
+        if (inputStream != null) {
+          inputStream.close();
+        }
+        if (outputStream != null) {
+          outputStream.close();
+        }
+      } catch (Exception e) {
+        // empty catch block. This is added incase exception occurs before releasing sharedLock.
       }
       sharedLock.unlock();
```
Comment (desaikomal) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorage.java:181]: you can use finally () block

**Review by desaikomal — COMMENTED** [review]
i am not sure where snapLog and timeTable<> class variables used as i didn't see any accessor methods as well.

File: zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorage.java [hunk 5]
```diff
@@ -166,11 +168,15 @@ public void copyToBackupStorage(File srcFile, File destName) throws IOException
           Paths.get(BackupStorageUtil.constructBackupFilePath(destName.getName(), fileRootPath)),
           StandardCopyOption.REPLACE_EXISTING);
     } finally {
-      if (inputStream != null) {
-        inputStream.close();
-      }
-      if (outputStream != null) {
-        outputStream.close();
+      try {
```
Comment (mgao0) [inline zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/impl/FileSystemBackupStorage.java:171]: Could you please help me understand what error is it in this place? Why do we want try-catch inside finally block?

**Review by mgao0 — COMMENTED** [review]
Thanks for working on fixing the errors! Overall looks good, left one comment. Great work!

**Comment by rahulrane50** [comment]
Sorry for the mess i will delete this PR and create new PR. I will address all these comments in new PR

**Pull request closed without merging**